string control
---
bool is_driving
int64 max_time
